collector/cpu_aix.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nocpu
+
+package collector
+
+/*
+#cgo LDFLAGS: -lperfstat
+#include <stdlib.h>
+#include <stdio.h>
+#include <libperfstat.h>
+#include <string.h>
+#include <time.h>
+
+u_longlong_t **ref;
+
+int getCPUTicks(uint64_t **cputicks, size_t *cpu_ticks_len) {
```

Comment:
Rather than implement this in C, why not call `C.perfstat_cpu()` from a go function?

Reply:
I am following what we do for https://github.com/prometheus/node_exporter/blob/7ad86f7994d6f1f290a28fcc2e16e54193e5ab9e/collector/cpu_dragonfly.go#L109 today. Is it better to move this to different cgo calls?

Reply:
Sorry for the delay in reviewing. this makes sense.